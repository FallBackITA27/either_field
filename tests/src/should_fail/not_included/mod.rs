
#[either_field::make_template(
    NotIncluded:
    [
        either_type_4: &'static str
    ],
)]
struct GenericStruct<A: IntoIterator, B: std::fmt::Debug + 'static> {
    generic_a: A,
    generic_b: B,
    either_type_0: either_field::either!(i32 | &'static str),
    either_type_1: either_field::either!(i32 | &'static str | u32),
    either_type_2: either_field::either!(i32 | &'static str | String),
    either_type_3: either_field::either!(i32 | A | String),
    either_type_4: either_field::either!(i32 | (A, B) | String),
}


fn main() {}
