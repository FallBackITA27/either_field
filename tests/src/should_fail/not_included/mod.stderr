error: Type "& 'static str" is not part of the specified possible types: ["i32", "(A, B)", "String"]
 --> src/should_fail/not_included.rs:2:1
  |
2 | / #[either_field::make_template(
3 | |     NotIncluded:
4 | |     [
5 | |         either_type_4: &'static str
6 | |     ],
7 | | )]
  | |__^
  |
  = note: this error originates in the attribute macro `either_field::make_template` (in Nightly builds, run with -Z macro-backtrace for more info)
